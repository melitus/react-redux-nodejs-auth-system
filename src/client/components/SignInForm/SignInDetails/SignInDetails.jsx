// @flow
import React from 'react';
import { Field, FormSection } from 'redux-form';

import { Input } from '../../Field/index';

type SignInFormDetailsProp = {
  password?: number,
  email: string,
};

const SignInDetails = ({ email, password }: SignInFormDetailsProp) => (
  <FormSection name="signInDetails">
    <label target="signin.signInDetails.email">Email</label>
    <Field
      id="signin.personInfo.email"
      name="email"
      component={Input}
      placeholder="Email"
      type="email"
      value={email}
    />

    <label target="signin.signInDetails.passwword">Password</label>
    <Field
      id="signup.signinDetail.password"
      name="password"
      component={Input}
      placeholder="••••••••••"
      type="password"
      value={password}
    />
  </FormSection>
);

SignInDetails.defaultProps = {
  email: '',
  password: '',
};

export default SignInDetails;
