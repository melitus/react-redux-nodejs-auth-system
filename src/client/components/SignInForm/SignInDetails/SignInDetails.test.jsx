import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

import SignInDetails from './SignInDetails';

enzyme.configure({ adapter: new Adapter() });

describe('<SignInDetails />', () => {
  it('should render default snapshot', () => {
    const wrapper = shallow(
      <SignInDetails />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
