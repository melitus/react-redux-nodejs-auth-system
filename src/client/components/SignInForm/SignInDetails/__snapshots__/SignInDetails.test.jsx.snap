exports[`<SignInDetails /> should render default snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignInDetails
    email=""
    password="" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label
          target="signin.signInDetails.email">
          Email
</label>,
        <Field
          component={undefined}
          id="signin.personInfo.email"
          name="email"
          placeholder="Email"
          type="email"
          value="" />,
        <label
          target="signin.signInDetails.passwword">
          Password
</label>,
        <Field
          component={undefined}
          id="signup.signinDetail.password"
          name="password"
          placeholder="••••••••••"
          type="password"
          value="" />,
      ],
      "component": "div",
      "name": "signInDetails",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Email",
          "target": "signin.signInDetails.email",
        },
        "ref": null,
        "rendered": "Email",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": undefined,
          "id": "signin.personInfo.email",
          "name": "email",
          "placeholder": "Email",
          "type": "email",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Password",
          "target": "signin.signInDetails.passwword",
        },
        "ref": null,
        "rendered": "Password",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": undefined,
          "id": "signup.signinDetail.password",
          "name": "password",
          "placeholder": "••••••••••",
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label
            target="signin.signInDetails.email">
            Email
</label>,
          <Field
            component={undefined}
            id="signin.personInfo.email"
            name="email"
            placeholder="Email"
            type="email"
            value="" />,
          <label
            target="signin.signInDetails.passwword">
            Password
</label>,
          <Field
            component={undefined}
            id="signup.signinDetail.password"
            name="password"
            placeholder="••••••••••"
            type="password"
            value="" />,
        ],
        "component": "div",
        "name": "signInDetails",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email",
            "target": "signin.signInDetails.email",
          },
          "ref": null,
          "rendered": "Email",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": undefined,
            "id": "signin.personInfo.email",
            "name": "email",
            "placeholder": "Email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Password",
            "target": "signin.signInDetails.passwword",
          },
          "ref": null,
          "rendered": "Password",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": undefined,
            "id": "signup.signinDetail.password",
            "name": "password",
            "placeholder": "••••••••••",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
